subprojects/ide/src/integTest/groovy/org/gradle/plugins/ide/AbstractIdeDeduplicationIntegrationTest.groovy

```diff
@@ -257,6 +259,52 @@ abstract class AbstractIdeDeduplicationIntegrationTest extends AbstractIdeIntegr
         projectName("impl/myproject/myproject-foo/app") == "impl-myproject-foo-app"
     }
 
+    def "will use configured module name"() {
+        given:
+        project("root") {
+            project("foo") {
+                project("app") {
+                    buildFile << "${configuredModule}.name = 'custom-app'"
+                }
+            }
+            project("bar") {
+                project("app") {}
+            }
+        }
+
+        when:
+        run ideName
+
+        then:
+        projectName("foo") == "foo"
+        projectName("foo/app") == "custom-app"
+        projectName("bar") == "bar"
+        projectName("bar/app") == "bar-app"
+    }
+
+    def "will not de-duplicate module that conflicts with configured module name"() {
+        given:
+        project("root") {
+            project("foo") {
+                project("other") {
+                    buildFile << "${configuredModule}.name = 'app'"
+                }
+            }
+            project("bar") {
+                project("app") {}
+            }
+        }
+
+        when:
+        run ideName
+
+        then:
+        projectName("foo") == "foo"
+        projectName("foo/other") == "app"
+        projectName("bar") == "bar"
+        projectName("bar/app") == "app"
```

Comment:
Yes, this will result in a badly formed IDE project.  I think it's OK for this to happen when the user explicitly sets the duplicate name. Previously, we attempted to de-duplicate even the names that were provided, but this was error prone and resulted in very tangled code.

I'm tempted to leave it like this.